proto/tendermint/types/types.proto

```diff
@@ -89,18 +89,12 @@ message Data {
   // NOTE: not all txs here are valid. We're just agreeing on the order first.
   repeated bytes txs = 1;
 
-  reserved 2, 3, 4;
+  reserved 2, 3, 4, 5, 6;
   // field number 2 is reserved for intermediate state roots
   // field number 3 is reserved for evidence
   // field number 4 is reserved for blobs
-
-  // SquareSize is the number of rows or columns in the original data square.
-  uint64 square_size = 5;
-
-  // Hash is the root of a binary Merkle tree where the leaves of the tree are
-  // the row and column roots of an extended data square. Hash is often referred
-  // to as the "data root".
-  bytes hash = 6;
```

Comment:
I think we need to keep hash here unless we want to modify some of the validation rules in `state/validation.go`